### DISCUSSION TITLE
Failed to select streams for reading

### DISCUSSION THREAD

**Discussion by s4muel** [sirn-se/websocket-php#76] [Q&A] [answered]
hello everyone. i have a strange error i want to discuss about. this is my setup:

* web (react with laravel echo) sends a http request to my laravel backend
* laravel job is created
* the job is processed by laravel horizon (queue worker)
* the job is a simple websocket client that connects to a WS stream, the stream sends like 100 text messages over 50-70 seconds, that i handle with `onText()` listener
* on every text a new event (job) is fired and added to the queue (so around 100 jobs) to be later process by the horizon again to push (laravel reverb) to the frontend where echo is listening

(might be designed better, but it is what it is:)

when i run the client just to echo the messages or do not dispatch those ~100 jobs, everything works fine. but when they are fired, despite the processing seems to be smooth, all jobs are dispatched, processed... but from time to time i get `Failed to select streams for reading` exception:
```
ErrorException: stream_select(): Unable to select [4]: Interrupted system call (max_fd=22) in /var/www/project/vendor/phrity/net-stream/src/StreamCollection.php:119
Stack trace:
#0 [internal function]: Phrity\Util\ErrorHandler->Phrity\Util\{closure}(2, 'stream_select()...', '/var/www/project/ve...', 119)
#1 /var/www/project/vendor/phrity/net-stream/src/StreamCollection.php(119): stream_select(Array, Array, Array, 60)
#2 /var/www/project/vendor/phrity/util-errorhandler/src/ErrorHandler.php(60): Phrity\Net\StreamCollection->Phrity\Net\{closure}()
#3 /var/www/project/vendor/phrity/net-stream/src/StreamCollection.php(117): Phrity\Util\ErrorHandler->with(Object(Closure), Object(Phrity\Net\StreamException))
#4 /var/www/project/vendor/phrity/websocket/src/Client.php(275): Phrity\Net\StreamCollection->waitRead(60)
#5 /var/www/project/app/Jobs/TestJob.php(100): WebSocket\Client->start()
```
i googled around, asked chatgpt... but i am stuck. only issue that i've found, that resembles mine, is this one: https://github.com/PHPMailer/PHPMailer/issues/1227

do some of you have any idea or a suggestion i could do do fix the issue?

edit: in the phpmailer issue above discussion, there was a link to https://github.com/reactphp/event-loop/pull/44/files
as mentioned there, i also try to suppress the warnings with an `@` here: https://github.com/sirn-se/phrity-net-stream/blob/main-v2/src/StreamCollection.php#L119 and it seems to be working fine since then. not saying it is the right solution, just letting you my observation

**Comment by sirn-se** [comment, marked as answer]
Googling the warning message, it seams this is something that occasionally happens and cannot be circumvented. All implementations I have looked at handles or ignores the error in some way.

So basically I need to fix this by treating it as an "expected issue", and not throwing an Exception as of now.

**Reply by s4muel** [reply]
thank you for the quick reply. looking forward to it.

**Comment by sirn-se** [comment]
I have updated the underlying `phrity/net-stream` library as `v2.1` to handle this situation.
`v3.2` of this repo enforces this update.

**Reply by s4muel** [reply]
thanks
